src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/ApplicationTest.kt

```diff
@@ -3408,61 +3439,64 @@ class ApplicationTest : IntegrationTestBase() {
               .isOk
               .expectBody()
-              .json(objectMapper.writeValueAsString(expected))
+              .jsonForObject(expected)
           }
         }
       }
     }
 
     @Test
     fun `Get withdrawables for all possible types filters out bookings when a user cannot manage bookings`() {
-      `Given a User` { applicant, _ ->
-        `Given a User` { user, jwt ->
-          `Given an Offender` { offenderDetails, _ ->
+      `Given a User` { applicant, jwt ->
+        `Given an Offender` { offenderDetails, _ ->
 
-            val (application, _) = produceAndPersistApplicationAndAssessment(applicant, user, offenderDetails)
+          val (application, _) = produceAndPersistApplicationAndAssessment(applicant, applicant, offenderDetails)
 
-            val booking1ExpectedArrival = LocalDate.now().plusDays(1)
-            val booking1ExpectedDeparture = LocalDate.now().plusDays(6)
-            produceAndPersistBooking(
-              application,
-              booking1ExpectedArrival,
-              booking1ExpectedDeparture,
-            )
+          val booking1ExpectedArrival = LocalDate.now().plusDays(1)
+          val booking1ExpectedDeparture = LocalDate.now().plusDays(6)
+          produceAndPersistBooking(
+            application,
+            booking1ExpectedArrival,
+            booking1ExpectedDeparture,
+          )
 
-            val placementApplicationExpectedArrival = LocalDate.now().plusDays(1)
-            val placementApplicationDuration = 5
+          val placementApplicationExpectedArrival = LocalDate.now().plusDays(1)
+          val placementApplicationDuration = 5
 
-            val placementApplication = produceAndPersistPlacementApplication(
-              application,
-              listOf(placementApplicationExpectedArrival to placementApplicationDuration),
-            )
+          val placementApplication = produceAndPersistPlacementApplication(
+            application,
+            listOf(placementApplicationExpectedArrival to placementApplicationDuration),
+          )
 
-            val placementRequest = produceAndPersistPlacementRequest(application)
+          val placementRequest = produceAndPersistPlacementRequest(application)
 
-            val expected = listOf(
-              Withdrawable(
-                placementApplication.id,
-                WithdrawableType.placementApplication,
-                listOf(datePeriodForDuration(placementApplicationExpectedArrival, placementApplicationDuration)),
-              ),
-              Withdrawable(
-                placementRequest.id,
-                WithdrawableType.placementRequest,
-                listOf(datePeriodForDuration(placementRequest.expectedArrival, placementRequest.duration)),
-              ),
-            )
+          val expected = listOf(
+            Withdrawable(
+              application.id,
+              WithdrawableType.application,
+              emptyList(),
+            ),
```

Comment:
Do you agree that this PR could be easier to review (i.e. easier for the reviewer to spot issues / misunderstandings) if the change to behaviour were isolated in a single commit, separate from the various helpful refactorings? e.g.

- preliminary commit to fixup existing formatting
- commit to reformat `integration/ApplicationTest.kt`, whitespace, renaming block variables (`user` becomes `allocatedTo`)
- commit to introduce `jsonForObject` helper
- commit to change behaviour of `/withdrawables` endpoint to include new `application` type

Reply:
i've moved the jsonForObject change to its own commit as I agree it shouldn't be mixed in with the new functionality

As for formatting, alot of the whitespace change was introduced because i had to add an additional 'Given a User' block into the test for the change. If we moved away from using lots of embedded code blocks that would massively reduce noise for changes like this, but most of the code is done this way unfortunately

As for variable name change from 'user' to 'allocatedTo', i agree it probably should be on its own commit but i feel like small variable names should be ok in 'unrelated' commits as the change is pretty clear? I'll try to do such changes on a separate commit going forward.

Reply:
👍 thks David, I think I failed to appreciate that some of the whitespace/indentation changes were related to that additional Given block! apologies 🙇